### PR TITLE
GD-31:  Adding spy and mock on scenes.

### PR DESCRIPTION
**Pull request by MikeSchulze**
- added example MenuDemo2D

### CODE REVIEW

File: addons/gdUnit3/src/mocking/GdUnitMockBuilder.gd [hunk 1]
```diff
@@ -161,19 +161,58 @@ static func build(clazz, mock_mode :String, memory_pool :int, debug_write = fals
...
+	var script_path = scene_instance.get_script().get_path()
+	var mock = mock_on_script(script_path, GdUnitMock.CALL_REAL_FUNC, memory_pool, GdUnitClassDoubler.EXLCUDE_SCENE_FUNCTIONS, debug_write)
+	if mock == null:
+		return null
+	scene_instance.set_script(mock)
+	scene_instance.__set_singleton()
+	scene_instance.__set_mode(GdUnitMock.CALL_REAL_FUNC)
+	return GdUnitTools.register_auto_free(scene_instance, memory_pool)
+
+static func mock_on_script(clazz, mock_mode :String, memory_pool :int, function_excludes :PoolStringArray, debug_write :bool) -> GDScript:
```
Comment (MikeSchulze) [inline addons/gdUnit3/src/mocking/GdUnitMockBuilder.gd:200]: function_excludes :Array

**Comment by MikeSchulze** [comment]
Hi @afonsolage  @clemens-tolboom the scene mock/spy is mostly done ;)
But to finallize i have to fix runtime errors on GdUnitSceneRunnerTest.gd 
It shows some errors related to yielding stuff, but i have current no idea what goes wrong here.
When i execute each test as single no errors occurs but when i run the complete testsuite than this strange errors pops up.

**Comment by clemens-tolboom** [comment]
@MikeSchulze I'm not sure I know what a Spy is :-p ... I hope to (re)start my project using this PR soon :-/

**Comment by MikeSchulze** [comment]
> @MikeSchulze I'm not sure I know what a Spy is :-p ... I hope to (re)start my project using this PR soon :-/

have a look here https://github.com/MikeSchulze/gdUnit3/wiki/Spy#definition
The main difference is that you work on a full initialized object the real implementation is called.

**Pull request merged**